lib/cli/src/commands/init.rs

```diff
@@ -0,0 +1,472 @@
+            wasmer_extra_flags: None,
+            abi: default_abi,
+            fs: constructed_manifest.fs,
+            bindings,
+        };
+
+        let toml_string = toml::to_string_pretty(&metadata_wapm)?
+            .replace("[dependencies]", &format!("{note}\r\n\r\n[dependencies]"))
+            .lines()
+            .collect::<Vec<_>>()
+            .join("\r\n");
+
+        if !self.quiet {
+            eprintln!(
+                "You have cargo-wapm installed, added metadata to Cargo.toml instead of wasmer.toml"
+            );
+            eprintln!("Build and publish your package with:");
+            eprintln!();
+            eprintln!("    cargo wapm");
+            eprintln!();
+        }
+
+        std::fs::write(
+            &manifest_path,
+            &format!("{old_cargo}\r\n\r\n[package.metadata.wapm]\r\n{toml_string}"),
+        )?;
+
+        Ok(())
+    }
+
+    fn get_filesystem_mapping(&self) -> Option<HashMap<String, PathBuf>> {
+        if self.include.is_empty() {
+            return None;
+        }
+
+        let include = self
+            .include
+            .iter()
+            .map(|path| {
+                if path == "." || path == "/" {
+                    return ("/".to_string(), Path::new("/").to_path_buf());
+                }
+
+                let key = format!("./{path}");
+                let value = Path::new(&format!("/{path}")).to_path_buf();
+
+                (key, value)
+            })
+            .collect();
+
+        Some(include)
+    }
+
+    fn get_command(
+        modules: &[wapm_toml::Module],
+        bin_or_lib: BinOrLib,
+    ) -> Option<Vec<wapm_toml::Command>> {
+        match bin_or_lib {
+            BinOrLib::Bin => Some(
+                modules
+                    .iter()
+                    .map(|m| {
+                        wapm_toml::Command::V1(wapm_toml::CommandV1 {
+                            name: m.name.clone(),
+                            module: m.name.clone(),
+                            main_args: None,
+                            package: None,
+                        })
+                    })
+                    .collect(),
+            ),
+            BinOrLib::Lib | BinOrLib::Empty => None,
+        }
+    }
+
+    /// Returns the dependencies based on the `--template` flag
+    fn get_dependencies(&self) -> Option<HashMap<String, String>> {
+        Some({
+            match self.template.as_deref() {
+                Some("js") => {
+                    let mut map = HashMap::default();
+                    map.insert("python".to_string(), "quickjs/quickjs@latest".to_string());
+                    map
+                }
+                Some("python") => {
+                    let mut map = HashMap::default();
+                    map.insert("python".to_string(), "python/python@latest".to_string());
+                    map
+                }
+                _ => HashMap::default(),
+            }
+        })
+    }
```

Comment:
We can probably get rid of some nesting here:

```suggestion
    fn get_dependencies(&self) -> Option<HashMap<String, String>> {
        let mut map = HashMap::new();
        
        match self.template.as_deref() {
            Some("js") => {
                map.insert("python".to_string(), "quickjs/quickjs@latest".to_string());
            }
            Some("python") => {
                map.insert("python".to_string(), "python/python@latest".to_string());
            }
            _ => {},
            }
            
            Some(map)
    }
```